src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/PreviewAPIListBuilder.java

```diff
@@ -60,40 +60,51 @@ public int compareTo(JEP o) {
      * @param configuration the current configuration of the doclet
      */
     public PreviewAPIListBuilder(BaseConfiguration configuration) {
-        super(configuration, configuration.utils::isPreviewAPI);
+        super(configuration);
         buildSummaryAPIInfo();
     }
 
     @Override
-    protected void handleElement(Element e) {
-        String feature = Objects.requireNonNull(utils.getPreviewFeature(e),
+    protected boolean belongsToSummary(Element element) {
+        if (!utils.isPreviewAPI(element)) {
+            return false;
+        }
+        String feature = Objects.requireNonNull(utils.getPreviewFeature(element),
                 "Preview feature not specified").toString();
-        JEP jep = jeps.computeIfAbsent(feature, (featureName) -> {
-            Map<? extends ExecutableElement, ? extends AnnotationValue> anno = configuration.workArounds.getJepInfo(featureName);
-            int number = 0;
-            String title = "";
-            String status = "Preview"; // Default value is not returned by the method we use above.
-            for (var entry : anno.entrySet()) {
-                if ("number".equals(entry.getKey().getSimpleName().toString())) {
-                    number = (int) entry.getValue().getValue();
-                } else if ("title".equals(entry.getKey().getSimpleName().toString())) {
-                    title = (String) entry.getValue().getValue();
-                } else if ("status".equals(entry.getKey().getSimpleName().toString())) {
-                    status = (String) entry.getValue().getValue();
-                } else {
-                    throw new IllegalArgumentException(entry.getKey().getSimpleName().toString());
+        JEP jep = jeps.computeIfAbsent(feature, featureName -> {
+            Map<String, Object> jepInfo = configuration.workArounds.getJepInfo(featureName);
+            if (!jepInfo.isEmpty()) {
+                int number = 0;
+                String title = "";
+                String status = "Preview"; // Default value is not returned by the method we used above.
+                for (var entry : jepInfo.entrySet()) {
+                    switch (entry.getKey()) {
+                        case "number" -> number = (int) entry.getValue();
+                        case "title" -> title = (String) entry.getValue();
+                        case "status" -> status = (String) entry.getValue();
+                        default -> throw new IllegalArgumentException(entry.getKey());
+                    }
                 }
+                return new JEP(number, title, status);
             }
-            return new JEP(number, title, status);
+            return NULL_SENTINEL;
         });
-        elementJeps.put(e, jep);
+        if (jep != NULL_SENTINEL) {
+            elementJeps.put(element, jep);
+            return true;
+        }
+        // Preview features without JEP are not included.
```

Comment:
Trivial: we don't need to mention "support" near "preview" and "features", right? Just checking.